### PR TITLE
fix(chan): postpone trimming conninfo after `connected` hook run 

### PR DESCRIPTION
**Pull request by keynslug**
Fixes [EMQX-9897](https://emqx.atlassian.net/browse/EMQX-9897)

### <samp>🤖 Generated by Copilot at dc32697</samp>

Remove peercert field from channel and conninfo structures to save memory. Update type definitions and test cases accordingly. Refactor and simplify some test code.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] ~~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~~
- [x] Schema changes are backward compatible

### CODE REVIEW

File: changes/ce/fix-10715.en.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Postpone trimming the connection information structure until after any `client.connected` hooks have been executed. These hooks once again have access to the client's peer certificate.
```
Comment (zmstone) [inline changes/ce/fix-10715.en.md:1]:
```suggestion
Postpone trimming the connection information structure until after `client.connected` hooks have been executed. These hooks once again have access to the client's peer certificate.
```

File: apps/emqx/src/emqx_channel.erl [hunk 2]
```diff
@@ -1989,10 +1987,21 @@ ensure_connected(
...
+        conninfo = trim_conninfo(NConnInfo),
         conn_state = connected
     }.
 
+trim_conninfo(ConnInfo) ->
+    maps:without(
+        [
+            %% NOTE
+            %% We remove the peercert because it duplicates what's stored in the socket,
+            %% unnecessary wasting memory (about 1KB per connection).
```
Comment (zmstone) [inline apps/emqx/src/emqx_channel.erl:1999]:
```suggestion
            %% otherwise it wastes about 1KB per connection.
```

**Review by zmstone — APPROVED** [review]

**Comment by yanzhiemq** [comment]
_

**Comment by yanzhiemq** [comment]
_

**Pull request merged**